metricflow/model/data_warehouse_model_validator.py

```diff
@@ -41,86 +76,82 @@ class DataWarehouseValidationTask:
+            DataSourceReference(data_source_name=data_source.name)
+        )
+        assert data_source_semantics
 
-    @staticmethod
-    def _where_clause_from_partitions(data_source: DataSource) -> str:
-        """Takes the partitions for a data source and genrates a WHERE clause based on their definintions"""
-        if not data_source.partitions:
-            return ""
+        source_nodes = render_tools.source_node_builder.create_from_data_sets(
+            (render_tools.converter.create_sql_source_data_set(data_source_semantics),)
+        )
 
-        where_stmts: List[str] = []
-        for partition in data_source.partitions:
-            element = f"({partition.expr})" if partition.expr else partition.name
-            where_stmts.append(DataWarehouseTaskBuilder.PARTITION_COL_TEMPLATE.format(partition=element))
-        return DataWarehouseTaskBuilder.WHERE_TEMPLATE.format(where_filters=" AND ".join(where_stmts))
+        assert len(source_nodes) == 1
+        return source_nodes[0]
 
     @staticmethod
-    def _gen_query(data_source: DataSource, id: int, columns: List[str] = ["true"]) -> str:
-        """Generates a basic sql query to select parts of them model definition
-
-        Generates a basic sql query for verifying the model's definition with
-        the data warehouse. For example if a data source with an sql_table
-        value of 'table1' and no partition was passed in no columns list the
-        query generated would be "SELECT (true) as col1 FROM table1 AS source0". If a the
-        data source had a partition, say on the dimension 'dim1' then the query
-        would be "SELECT (true) as col1 FROM table1 AS source0 WHERE dim1 IS NOT NULL".
-        And if in addition columns ["dim2", "dim3"] were passed in then the
-        query would be "SELECT (dim2) as col1, (dim3) as col2 FROM table1 AS source0 WHERE
-        dim1 IS NOT NULL".
-
-        Args:
-            data_source: The data source to generate the query for
-            id: A unique id used to alias the table reference
-            columns: Column strings to select
-        Returns:
-            A string representing the query for the passed in arguments.
-        """
-        data_source_str = data_source.sql_table if data_source.sql_table else f"({data_source.sql_query})"
-
-        wrapped_cols = [DataWarehouseTaskBuilder._wrap_col(col, index) for index, col in enumerate(columns)]
-        columns_select = ", ".join(wrapped_cols)
-
-        query = DataWarehouseTaskBuilder.QUERY_TEMPLATE.format(
-            data_source=data_source_str, columns_select=columns_select, unique_id=id
+    def renderize(
+        sql_client: SqlClient, plan_converter: DataflowToSqlQueryPlanConverter, plan_id: str, nodes: FilterElementsNode
+    ):
+        """Generates a sql query plan and returns the rendered sql and execution_parameters"""
+        sql_plan = plan_converter.convert_to_sql_query_plan(
```

Comment:
Looking at the old class and what we're doing in here it seems like we're running a lot of fairly simple queries. Shall we add a `limit 1` here to speed things up on the final stage? For the ones that just grab a column this will be a lot faster, and for anything aggregating a metric it'll have to compute the full set and then truncate the result.

Check out the OrderByLimitNode for this - you can pass an empty list for order by specs and then it'll just process the limit. You might need to play around with the node ordering to avoid selecting other columns in the inner query (also you need full optimization, otherwise we will select everything in the data source by name).

This is probably best left for a separate PR, since the existing queries do not use the LIMIT statement and this is a refactor, but it's a thought.

Reply:
So when we actually go about running the query we use the engine's implementation of dry run (https://github.com/transform-data/metricflow/blob/47ad03a9cfd12a5de36228784480cbb46589c0b0/metricflow/model/data_warehouse_model_validator.py#L301). For most engines this means that an `EXPLAIN` is appended to the front of the query. I think the exception is BigQuery which instead appends `dry_run` to the data warehouse query URL.